Sources/performance-worker/WorkerService.swift

```diff
@@ -41,12 +51,41 @@ final class WorkerService: Grpc_Testing_WorkerService.ServiceProtocol, Sendable
       self.role = role
     }
 
-    init(server: GRPCServer) {
-      self.role = .server(server)
+    var server: GRPCServer? {
+      switch self.role {
+      case let .server(serverState):
+        return serverState.server
+      default:
+        return nil
+      }
     }
 
-    init(client: GRPCClient) {
-      self.role = .client(client)
+    mutating func serverStats(replaceWith newStats: ServerStats? = nil) -> ServerStats? {
+      switch self.role {
+      case var .server(serverState):
+        let stats = serverState.stats
+        if let newStats = newStats {
+          serverState.stats = newStats
+          self.role = .server(serverState)
+        }
+        return stats
+      default:
+        return nil
+      }
+    }
+
+    mutating func setupServer(server: GRPCServer, stats: ServerStats) throws {
+      let serverState = State.ServerState(server: server, stats: stats)
+      switch self.role {
+      case .server(_):
+        throw RPCError(code: .alreadyExists, message: "A server has already been set up.")
+
+      case .client(_):
+        throw RPCError(code: .failedPrecondition, message: "This worker has a client setup.")
+
+      default:
```

Comment:
Same here re: explicitly switching
```suggestion
      case .none:
```